tools/transformer/src/transformations.js

```diff
@@ -148,61 +139,74 @@ function transformAllOf(objectPath, api) {
 
 function transformOneOf(objectPath, api) {
   const parentObject = getObjectFromYamlPath(objectPath, api);
+  const parentName = getNameFromYamlPath(objectPath);
 
-  if (!(parentObject && parentObject.oneOf)) {
-    throw new Error(`Invalid object for OneOf Transformation: ${objectPath}`);
+  if (!parentObject.oneOf) {
+    throw new Error(`Invalid object for OneOf Transformation: ${parentName}`);
   }
 
-  for (const objRef of parentObject.oneOf) {
-    const oneOfObject = getObjectFromReference(objRef, api);
-    if (!oneOfObject) {
-      const errorData = JSON.stringify(oneOfObject);
-      throw new Error(`Invalid or missing reference: ${errorData}`);
-    }
+  // Expand references
+  const childObjects = parentObject.oneOf.map((childRef) =>
+    getObjectFromReference(childRef, api)
+  );
+  const isEnum = childObjects.reduce(
+    (isEnum, childObject) => isEnum && childObject.enum,
+    true
+  );
 
-    if (oneOfObject.properties && oneOfObject.enum) {
-      const errorData = JSON.stringify(oneOfObject);
-      throw new Error(
-        `Object cannot contain both enum and properties: ${errorData}`
-      );
-    }
+  if (isEnum) {
+    transformOneOfEnum(parentObject, api);
+  } else {
+    transformOneOfObject(parentObject, api);
```

Comment:
```suggestion
    transformOneOfProperties(parentObject, api);
```

Reply:
[IMHO we will never be able to run properties transformation out of the box as it will literally remove all oneOf from the code. Do you think we can experiment with the transformation to see how our SDKs will look like in that sense? How many duplicates we will get? Would UX will be affected etc.?